dev.c

```diff
@@ -210,18 +210,16 @@ static void request_end(struct fuse_conn *fc, struct fuse_req *req,
 	int status)
 {
 	u64 uid = req->in.unique;
-#ifdef __PX_BLKMQ__
-	bool using_blkque = req->pxd_dev->using_blkque;
-#endif
+	bool shouldfree = false;
+
+	/* 'req->end' is always set if the context gets used.
+	 * if error happens during processing, error path handling does free request.
+	 */
 	if (req->end)
-		req->end(fc, req, status);
+		shouldfree = req->end(fc, req, status);
 	fuse_put_unique(fc, uid);
 
-#ifndef __PX_BLKMQ__
-	fuse_request_free(req);
-#else
-	if (!using_blkque) fuse_request_free(req);
-#endif
+	if (shouldfree) fuse_request_free(req);
```

Comment:
Instead of this, can we not do something like a safe_free(req)?

Reply:
All caller points are clearly defined and extended function about freeing fuse_request is unambiguous.
Whats the concern? And what is safe_free?

Reply:
I meant this:
```
	if (req->end)
		req->end(fc, req, status);
	fuse_put_unique(fc, uid);
        if (req)
                fuse_request_free(req);
...

#ifndef __PX_BLKMQ__
	blk_end_request(req->rq, status, blk_rq_bytes(req->rq));
#else
	blk_mq_end_request(req->rq, errno_to_blk_status(status));
	req = NULL;
#endif
```